pkg/controller/endpointslice/utils_test.go

```diff
@@ -73,18 +77,21 @@ func TestNewEndpointSlice(t *testing.T) {
 }
 
 func TestPodToEndpoint(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.EndpointSliceTerminatingCondition, true)()
```

Comment:
Can we make this something that applies per test case instead of for all of `TestPodToEndpoint`. Setting this to `true` here means we lose all our test coverage for the detault state where the feature gate is disabled. I think it would be helpful to leave the majority of existing tests covering that state and then add test cases that are specific to this feature. As an example, I think the default value for 1.20 is going to be nil but that is not actually covered by the tests here.

Reply:
Yup, good cal!